File: examples/C#/pathopub.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

using ZeroMQ;

namespace Examples
{
	static partial class Program
	{
		public static void PathoPub(string[] args)
		{
			//
			// Pathological publisher
			// Sends out 1,000 topics and then one random update per second
			//
			// Author: metadings
			//

			if (args == null || args.Length < 1)
			{
				Console.WriteLine();
				Console.WriteLine("Usage: ./{0} PathoPub [Endpoint]", AppDomain.CurrentDomain.FriendlyName);
				Console.WriteLine();
				Console.WriteLine("    Endpoint  Where PathoPub should connect to.");
				Console.WriteLine("              Default is null, Binding on tcp://*:5556");
				Console.WriteLine();
				args = new string[] { null };
			}

			using (var context = new ZContext())
			using (var publisher = new ZSocket(context, ZSocketType.PUB))
			{
				if (args[0] != null)
				{
					publisher.Connect(args[0]);
				}
				else
				{
					publisher.Bind("tcp://*:5556");
				}

				// Ensure subscriber connection has time to complete
				Thread.Sleep(100);

				// Send out all 1,000 topic messages
				for (int topic = 0; topic < 1000; ++topic)
				{
					publisher.SendMore(new ZFrame(string.Format("{0:D3}", topic)));
					publisher.Send(new ZFrame("Save Roger"));
				}

				// Send one random update per second
				var rnd = new Random();
				while (true)
				{
					Thread.Sleep(10);
					publisher.SendMore(new ZFrame(string.Format("{0:D3}", rnd.Next(1000))));
					publisher.Send(new ZFrame("Off with his head!"));
				}
			}
		}
	}
}

File: examples/Python/tornado_ioloop/pathopub.py
#!/usr/bin/env python

"""
synopsis:
    Pathological publisher
    Sends out 1,000 topics and then one random update per second
usage:
    python pathopub.py
"""

import sys
from random import randint
import zmq
from zmq.eventloop.future import Context
from zmq.eventloop.ioloop import IOLoop
from tornado import gen


Url = 'tcp://127.0.0.1:5555'
Ctx = Context()


@gen.coroutine
def run():
    publisher = Ctx.socket(zmq.PUB)
    publisher.bind(Url)
    # Ensure subscriber connection has time to complete
    yield gen.sleep(1)
    # Send out all 1,000 topic messages
    for topic_nbr in range(1000):
        yield publisher.send_multipart([
            b"%03d" % topic_nbr,
            b"Save Roger",
        ])
        print('.', end='')
    print()
    count = 0
    while True:
        count += 1
        # Send one random update per second
        yield gen.sleep(1)
        message = [
            b"%03d" % randint(0, 9),
            "Off with his head. {}".format(count).encode('utf-8'),
        ]
        yield publisher.send_multipart(message)
        print('sent message: {}'.format(message))


def main():
    args = sys.argv[1:]
    if len(args) != 0:
        sys.exit(__doc__)
    try:
        loop = IOLoop.current()
        loop.add_callback(lambda: run())
        loop.start()
    except KeyboardInterrupt:
        print('\nFinished (interrupted)')


if __name__ == '__main__':
    main()
